src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -1345,6 +1345,135 @@ class BSE(ExcitedStateMethodology):
     )
 
 
+class TDDFT(ExcitedStateMethodology):
+    """
+    Time-dependent density functional theory settings. Captures both linear-response
+    and real-time propagation flavours.
+
+    References
+    ----------
+    • E. Runge, E. K. U. Gross, Phys. Rev. Lett. 52, 997 (1984)  (TDDFT formalism)
+    • M. A. L. Marques et al. (eds.), *Fundamentals of Time-Dependent Density Functional Theory*,
+      Springer (2012)
+    • A. Castro et al., Phys. Status Solidi B 243, 2465 (2006)  (Real-time TDDFT overview)
+    """
+
+    type = Quantity(
+        type=MEnum('linear_response', 'real_time'),
+        default='linear_response',
+        description="""
+        TDDFT flavour (linear-response or real-time propagation).
+          • linear_response — frequency-domain (Casida, Sternheimer, Liouville-Lanczos)
+          • real_time       — explicit time propagation under a perturbation.
+        """,
+    )
+
+    kernel = Quantity(
+        type=MEnum(
+            'ALDA',
+            'AGGA',
+            'hybrid',
+            'range_separated',
+            'frequency_dependent',
+            'bootstrap',
+            'long_range_corrected',
+        ),
+        description="""
+        This field describes the effective XC response behaviour assumed in the TDDFT calculation, 
+        whether implemented explicitly (linear-response) or implicitly via the XC potential (real-time).
+          • ALDA / AGGA     : adiabatic local or semi-local
+          • hybrid          : hybrid adiabatic kernel
+          • range_separated : long/short-range split hybrids
+          • frequency_dependent : non-adiabatic kernel
+          • bootstrap / long_range_corrected : common nano/solid-state kernels
+        """,
+    )
+
+    is_tamm_dancoff = Quantity(
+        type=bool,
+        description='If True, Tamm-Dancoff approximation (TDA) is used (linear-response only).',
+    )
+
+    solver = Quantity(
+        type=MEnum(
+            'Casida', 'Sternheimer', 'Liouville-Lanczos', 'propagation', 'unavailable'
+        ),
+        description="""
+        Numerical driver:
+          • Casida             : eigenvalue problem in particle-hole space
+          • Sternheimer        : frequency-domain Sternheimer
+          • Liouville-Lanczos  : iterative Liouville-Lanczos
+          • propagation        : explicit time evolution (real-time)
+          • unavailable        : code-chosen / unspecified
+        """,
+    )
+
+    xc_kernel_ref = Quantity(
+        type=XCFunctional,
+        description="""
+        Reference to an `XCFunctional` describing the TDDFT kernel.
+        Typically points to the ground-state DFT XC section if the adiabatic
+        kernel matches, or to a separate XCFunctional created for the TD kernel
+        when it differs (e.g., LRC/hybrid kernel on top of semi-local ground state).
+        """,
+    )
```

Comment:
So, the DFT section has to be populated together with this one, correct? Maybe add this to the section's pydocs. (First time reading the description, I thought you were using the reference to describe a workflow, so the extra clarification would be nice.)
I would then also reference to `DFT` instead. Other corrections like Hubbard, Vandermonde, VdW corrections can all still be applied on top.